third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -562,7 +562,25 @@ void LoopPipeliner::emitEpilogue(DenseMap<Value, Value> &newResults) {
     epilogueMap.map(load.value()->getResult(0),
                     pplForOp.getResult(bufferIdx + load.index()));
   // Map IV to original upper bound (ie. last iteration)
-  epilogueMap.map(forOp.getInductionVar(), forOp.getUpperBound());
+
+  // This is computing the upper bound of the pipelined loop as:
+  //   pplUpperBound = lb+((ub- 1-lb)/step)*step
+  Location loc = forOp.getLoc();
+  Value ub = forOp.getUpperBound();
+  Value lb = forOp.getLowerBound();
+  Value step = forOp.getStep();
+  Value one = builder.create<arith::ConstantIntOp>(loc, 1, 32);
+  Value pplUpperBound = builder.create<arith::AddIOp>(
```

Comment:
Nit: can we create some local variables to hold the intermediate value? Currently this is formatted in a skewed manner. :)